File: src/main/java/com/todolist/task/service/application/mapper/CreateTaskCommandMapper.java
package com.todolist.task.service.application.mapper;

import com.todolist.task.service.adapter.http.models.CreateTaskCommand;
import com.todolist.task.service.domain.models.Task;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.ReportingPolicy;
import org.mapstruct.factory.Mappers;

@Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE)
public abstract class CreateTaskCommandMapper {

  public static final CreateTaskCommandMapper MAPPER = Mappers.getMapper(
      CreateTaskCommandMapper.class);


  @Mapping(target = "createdBy",source = "user")
  public abstract Task toDomain(CreateTaskCommand createTaskCommand , String user);

}

File: src/main/kotlin/com/todolist/task/service/application/mapper/CreateTaskCommandMapper.kt
package com.todolist.task.service.application.mapper

import com.todolist.task.service.application.models.CreateTaskCommand
import com.todolist.task.service.domain.enums.TaskStatus
import com.todolist.task.service.domain.models.Task
import java.time.Instant

abstract class CreateTaskCommandMapper {
    companion object {
        fun toDomain(command: CreateTaskCommand,user: String): Task {
            return Task(
                description = command.description,
                title = command.title,
                dueDate = command.dueDate,
                priority = command.priority,
                status = TaskStatus.TODO,
                createdBy = user,
                createdTime = Instant.now(),
                updatedBy = user,
                updatedTime = Instant.now()

            )
        }
    }
}
